### PR TITLE
Add media player screen to sample app

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add media player screen to sample app.

|Menu|Screen|
|---|---|
| Screenshot_20220509_102605 | Screenshot_20220509_112644 |

#### WHY

In order to have an example on how to build a screen with `PlayScreen` component.

#### HOW
- Add `MediaPlayerScreen` implementation using `PlayScreen`;
- Add `MediaPlayerChip` and add as an item in `MenuScreen` to navigate to `MediaPlayerScreen`;
- Add `MediaPlayerScreenViewModel` as a ViewModel for `MediaPlayerScreen`;
- Add `PlayerRepositoryImpl` as a fake implementation of `PlayerRepository`. It pretends that a list of `MediaItem`s are fetched and it pretends that they are playing when the play command is issued;
- Add `MediaDataSource` to provide hardcoded list of `MediaItem`s;
- Add `SampleAppDI`, a simple DI implementation until hilt is introduced to the project;
- Add `rememberStateWithLifecycle` utils function to sample app;

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: sample/src/main/java/com/google/android/horologist/sample/di/SampleAppDI.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+package com.google.android.horologist.sample.di
+
+import com.google.android.horologist.sample.MainActivity
+import com.google.android.horologist.sample.media.MediaDataSource
+import com.google.android.horologist.sample.media.MediaPlayerScreenViewModel
+import com.google.android.horologist.sample.media.PlayerRepositoryImpl
+
+/**
+ * Simple DI implementation - to be replaced by hilt.
```
Comment (yschimke) [inline sample/src/main/java/com/google/android/horologist/sample/di/SampleAppDI.kt:25]:
```suggestion
 * Simple DI implementation - to be replaced by hilt or Koin in a real app.
```

File: sample/src/main/java/com/google/android/horologist/sample/media/MediaPlayerScreen.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import androidx.compose.ui.Modifier
+import androidx.wear.compose.material.Scaffold
+import androidx.wear.compose.material.TimeText
+import com.google.android.horologist.media.ui.screens.PlayScreen
+
+@Composable
+fun MediaPlayerScreen(
+    mediaPlayerScreenViewModel: MediaPlayerScreenViewModel
+) {
+    Scaffold(
```
Comment (yschimke) [inline sample/src/main/java/com/google/android/horologist/sample/media/MediaPlayerScreen.kt:30]: As a follow up we should wire up the volume controls here, If you are busy, I'm happy to do it.

**Review by yschimke — APPROVED** [review]

File: sample/src/main/java/com/google/android/horologist/sample/media/MediaPlayerScreen.kt (hunk 2 above)
Comment (luizgrp) [reply sample/src/main/java/com/google/android/horologist/sample/media/MediaPlayerScreen.kt:30]: will add to this PR 👍

File: sample/src/main/java/com/google/android/horologist/sample/di/SampleAppDI.kt (hunk 1 above)
Comment (yschimke) [reply sample/src/main/java/com/google/android/horologist/sample/di/SampleAppDI.kt:25]: OK, I'll resolve this.

**Comment by yschimke** [comment]
LGTM

**Pull request merged**